pkg/common/directive.go

```diff
@@ -1,5 +1,10 @@
 package common
 
+type DirectiveVisitor interface {
```

Comment:
Maybe in some cases we need to access to the context at the directive level. What do you think to add `ctx context.Context` in `After` and `Before` definitions ?

Reply:
You're right, I just added the context

Reply:
One such case would be with a caching directive, since caching info can be stored in the context.